dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/MultipleBackJumpsToForwardJumps.java

```diff
@@ -0,0 +1,78 @@
+package com.dat3m.dartagnan.program.processing;
+
+import com.dat3m.dartagnan.program.Function;
+import com.dat3m.dartagnan.program.Program;
+import com.dat3m.dartagnan.program.event.Event;
+import com.dat3m.dartagnan.program.event.EventFactory;
+import com.dat3m.dartagnan.program.event.core.CondJump;
+import com.dat3m.dartagnan.program.event.core.Label;
+
+import static java.util.Comparator.comparingInt;
+
+import java.util.List;
+import java.util.function.ToIntFunction;
+
+/*
+    This pass transforms loops to have a single backjump using forward jumping.
+    Given a loop of the form
+
+    L:
+    ...
+    if X goto L
+    ...
+    if Y goto L
+    More Code
+
+    it transforms it to
+
+    L:
+    ...
+    if X goto Forward_to_L
+    ...
+    if Y goto Forward_to_L
+    goto Continue_after_last
+    Forward_to_L:
+    goto L
+    Continue_after_last
+    More Code
+    ...
+*/
+public class MultipleBackJumpsToForwardJumps implements ProgramProcessor {
+
+    public static MultipleBackJumpsToForwardJumps newInstance() {
+        return new MultipleBackJumpsToForwardJumps();
+    }
+
+    @Override
+    public void run(Program program) {
+        program.getFunctions().forEach(f -> backToForwardJumps(f, Event::getGlobalId));
+    }
+
+    private void backToForwardJumps(Function function, ToIntFunction<Event> linId) {
+        for (Label label : function.getEvents(Label.class)) {
+            final List<CondJump> backJumps = label.getJumpSet().stream()
+                    .filter(j -> linId.applyAsInt(j) > linId.applyAsInt(label))
+                    .toList();
+
+            if (backJumps.size() > 1) {
+                final CondJump last = backJumps.stream().max(comparingInt(linId)).get();
+
+                final Label forwardLabel = EventFactory.newLabel("Forward_to_" + label.getGlobalId());
+                final CondJump gotoHead = EventFactory.newGoto(label);
+
+                final Label continueLabel = EventFactory.newLabel("Continue_after_" + last.getGlobalId());
+                final CondJump gotoContinue = EventFactory.newGoto(continueLabel);
```

Comment:
Add `__` prefixes to internally generated names. Also, I think you could call the labels: `__continueLoop` (or `__repeatLoop`) and `__breakLoop`, indicating their purposes (in fact, you didn't even have to use explicit gotos to get multiple backjumps: a `continue` in a normal loop is a backjump).
Lastly, using `globalId` in the names is a bad idea because it will immediately cause mismatches.
I typically use a counter and add `#<counter>` as a suffix to give unique names (technically we do not even need unique names, but it makes CTRL+F searches easier in the IR).